var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -0,0 +1,239 @@
+from spack import *
+import sys
+
+class Dealii(Package):
+    """C++ software library providing well-documented tools to build finite element codes for a broad variety of PDEs."""
+    homepage = "https://www.dealii.org"
+    url      = "https://github.com/dealii/dealii/releases/download/v8.4.0/dealii-8.4.0.tar.gz"
+
+    version('8.4.0', 'ac5dbf676096ff61e092ce98c80c2b00')
+    version('dev', git='https://github.com/dealii/dealii.git')
+
+    variant('mpi',      default=True,  description='Compile with MPI')
+    variant('arpack',   default=True,  description='Compile with Arpack and PArpack (only with MPI)')
+    variant('doxygen',  default=False, description='Compile with Doxygen documentation')
```

Comment:
I think that variants like this are better spelled `+doc` since users, generally, won't care what generate the docs, but that they are generated.